sklearn/neighbors/kde.py

```diff
@@ -86,9 +92,14 @@ def __init__(self, bandwidth=1.0, algorithm='auto',
         # we're using clone() in the GenerativeBayes classifier,
         # so we can't do this kind of logic in __init__
         self._choose_algorithm(self.algorithm, self.metric)
+        
+        if bandwidth in ['scott', 'silvermann','cv']:
```

Comment:
Please set self.bandwidth in `__init__` with no validation. In fit, estimate and set `bandwidth_`, which you should use in score_samples